### PR TITLE
Add support for ponylang

### PR DESCRIPTION
**Pull request by mfelsche**
This add support for ponylang by adding this parser: https://github.com/mfelsche/tree-sitter-ponylang to the supported parsers, setting the revision, adding myself as the maintainer etc.

The queries seem legit. Highlighting works well. All queries have been tested in the tree-sitter playground and with nvim-treesitter as good as possible.

### CODE REVIEW

File: queries/ponylang/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  (block_comment)
+] @comment
+
+(bool) @boolean
+(integer) @number
+(float) @float
+(character) @character
+
+;; variables
+;; references to upper case things are considered constructors
```
Comment (ObserverOfTime) [inline queries/ponylang/highlights.scm:15]:
```suggestion
;; upper case identifiers are considered types
```

File: queries/ponylang/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+ "match"
+] @conditional
+[
+  "break"
+  "continue"
+  "return"
+  "error"
+  "compile_error"
+  "compile_intrinsic"
+] @keyword.return
```
Comment (ObserverOfTime) [inline queries/ponylang/highlights.scm:79]: I think only "return" should be here.

File: queries/ponylang/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+[
+  "recover"
+  "consume"
+  "end"
+  "try"
+  "with"
+] @keyword.control
```
Comment (ObserverOfTime) [inline queries/ponylang/highlights.scm:86]: Move these to the `@keyword` group.

File: queries/ponylang/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  "fun"
+  "be"
+  "new"
+] @keyword.function
+
+[
+  "var"
+  "let"
+  "embed"
+] @storageclass
```
Comment (ObserverOfTime) [inline queries/ponylang/highlights.scm:111]: "var" and "let" should be `@keyword`.

File: queries/ponylang/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+  "digestof"
+  "addressof"
+  (location)
+] @keyword.operator
+
+[
+  "fun"
+  "be"
+  "new"
+] @keyword.function
```
Comment (ObserverOfTime) [inline queries/ponylang/highlights.scm:105]: "be" should be `@keyword.coroutine`.
"new" should be `@keyword.operator`.

File: queries/ponylang/textobjects.scm
Comment (ObserverOfTime) [inline queries/ponylang/textobjects.scm:1]: Text-objects go in [nvim-treesitter-textobjects](https://github.com/nvim-treesitter/nvim-treesitter-textobjects).

File: lockfile.json [hunk 6]
```diff
@@ -338,6 +338,9 @@
   "poe_filter": {
     "revision": "80dc10195e26c72598ed1ab02cdf2d8e4c792e7b"
   },
+  "ponylang": {
+    "revision": "v0.0.2"
```
Comment (ObserverOfTime) [inline lockfile.json:342]: This should be the commit hash.

File: lua/nvim-treesitter/parsers.lua [hunk 7]
```diff
@@ -1085,6 +1085,17 @@ list.poe_filter = {
   readme_name = "Path of Exile item filter",
 }
 
+list.ponylang = {
+  install_info = {
+    url = "https://github.com/mfelsche/tree-sitter-ponylang",
+    revision = "v0.0.2",
```
Comment (ObserverOfTime) [inline lua/nvim-treesitter/parsers.lua:1091]: It should (normally) not be hardcoded here.

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

**Comment by mfelsche** [comment]
Thanks for the review @ObserverOfTime I am reworking the whole tree-sitter grammar with @amaanq so, this whole PR is going to change drastically. I will incoporate your remarks in that. Stay tuned.

**Pull request closed without merging**